### PR TITLE
Build source containers with remote_sources

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-4454

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 1]
```diff
@@ -219,40 +221,101 @@ def _get_remote_urls_helper(self, koji_build):
...
+                message = ''
+                if not archive_found:
+                    message = 'Remote sources archive'
+                if not json_found:
+                    if message:
+                        message += ' and remote sources json'
+                    else:
+                        message = 'Remote sources json'
+                message += ' missing.'
+                raise RuntimeError(message)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:260]: Can we have just something simpler, so much code just about formating proper error output

```python
message = ', 'join(part for t, part in (
                         (archive_found, "remote source archive missing"),
                         (json_found, "remote sources json missing")
                   ) if not t)
```

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 2]
```diff
@@ -219,40 +221,101 @@ def _get_remote_urls_helper(self, koji_build):
...
+                        remote_sources_urls.append(remote_source)
+                        cachito_json_url = os.path.join(remote_sources_path, remote_json)
+                        remote_json_map[remote_source['dest']] = cachito_json_url
+                        all_archives.add(remote_archive)
+                        all_archives.add(remote_json)
+
+            if wrong_archives:
+                raise RuntimeError('Problems with archives in remote sources: {}'.
+                                   format(remote_sources))
+
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:304]: Can be this extracted into separate method for better readability?

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 3]
```diff
@@ -219,40 +221,101 @@ def _get_remote_urls_helper(self, koji_build):
...
+                    if message:
+                        message += ' and remote sources json'
+                    else:
+                        message = 'Remote sources json'
+                message += ' missing.'
+                raise RuntimeError(message)
+
+            elif len(archives) > 2:
+                raise RuntimeError('There can be just one remote sources archive and one '
+                                   'remote sources json, got: {}'.format(all_archives))
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:264]: Can be this extracted into separate method for better readability?

**Review by MartinBasti — CHANGES_REQUESTED** [review]

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:260]: sure

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:264]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:304]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 4]
```diff
@@ -213,47 +215,118 @@ def assemble_srpm_url(self, base_url, srpm_filename, sign_key=None):
...
+                    remote_source['url'] = os.path.join(remote_sources_path, remote_archive)
+                    remote_source['dest'] = '-'.join([koji_build['nvr'], remote_archive])
+                    remote_sources_urls.append(remote_source)
+                    cachito_json_url = os.path.join(remote_sources_path, remote_json)
+                    remote_json_map[remote_source['dest']] = cachito_json_url
+                    all_archives.append(remote_archive)
+                    all_archives.append(remote_json)
+
+        if wrong_archives:
+            raise RuntimeError('Problems with archives in remote sources: {}'.
```
Comment (ben-alkov) [inline atomic_reactor/plugins/pre_fetch_sources.py:290]: Is it possible to have more details here re: "Problems"?

**Review by ben-alkov — APPROVED** [review]
LGTM with comment

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 4 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:290]: those are logged as errors above

**Review by mkosiarc — APPROVED** [review]

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 4 above)
Comment (ben-alkov) [reply atomic_reactor/plugins/pre_fetch_sources.py:290]: ACK, thx

**Review by MartinBasti — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated: Build source containers with remote_sources

**Pull request merged**